src/openrct2/peep/Peep.h

```diff
@@ -485,12 +484,19 @@ enum PeepRideDecision
     PEEP_RIDE_DECISION_THINKING = 1 << 2,
 };
 
-struct rct_peep_thought
+static constexpr uint16_t PeepThoughtItemNone = std::numeric_limits<uint16_t>::max();
+
+struct PeepThought
 {
-    PeepThoughtType type;  // 0
-    uint8_t item;          // 1
-    uint8_t freshness;     // 2 larger is less fresh
-    uint8_t fresh_timeout; // 3 updates every tick
+    PeepThoughtType type;
+    union
+    {
+        ride_id_t rideId;
+        ShopItem shopItem;
+        uint16_t item;
+    };
```

Comment:
Does this have to be a union? If size of the struct is not an issue, perhaps split them up or use `std::variant<ride_id_t, ShopItem, uint16_t>`. Import code likely needs to be changed then too.

Reply:
I thought about it but until ride_id_t is not a unique type this is error prone, at some point maybe.

Reply:
It would be a little hard to work out what each type is for the import.